src/main/java/uk/gov/ons/census/casesvc/messaging/SampleReceiver.java

```diff
@@ -9,75 +12,124 @@
+
+    Case caze = persistToDatabase(createCaseSample);
+
+    CaseCreatedEvent caseCreatedEvent = prepareEventToEmit(caze);
+
+    rabbitTemplate.convertAndSend(emitCaseEventExchange, "", caseCreatedEvent);
+  }
+
+  private Case persistToDatabase(CreateCaseSample createCaseSample) {
+    Case caze = mapperFacade.map(createCaseSample, Case.class);
+    caze.setCaseId(UUID.randomUUID());
+    caze.setState(CaseState.ACTIONABLE);
+    caze = caseRepository.saveAndFlush(caze);
+
+    UacQidLink uacQidLink = new UacQidLink();
+    uacQidLink.setId(UUID.randomUUID());
+    uacQidLink.setUac(iacDispenser.getIacCode());
+    uacQidLink.setCaze(caze);
+    uacQidLink = uacQidLinkRepository.saveAndFlush(uacQidLink);
+
+    int questionnaireType = 99;
+    int trancheIdentifier = 2;
+    uacQidLink.setQid(
+        qidCreator.createQid(questionnaireType, trancheIdentifier, uacQidLink.getUniqueNumber()));
+    uacQidLinkRepository.save(uacQidLink);
+
+    uk.gov.ons.census.casesvc.model.entity.Event loggedEvent =
+        new uk.gov.ons.census.casesvc.model.entity.Event();
+    loggedEvent.setId(UUID.randomUUID());
+    loggedEvent.setEventDate(new Date());
+    loggedEvent.setEventDescription("Case created");
+    loggedEvent.setUacQidLink(uacQidLink);
+    eventRepository.save(loggedEvent);
+
+    return caze;
+  }
+
+  private CaseCreatedEvent prepareEventToEmit(Case caze) {
+    LocalDateTime now = LocalDateTime.now();
 
     Event event = new Event();
     event.setChannel("rm");
-    event.setDateTime("2019-04-01T12:00Z");
+    event.setSource("CaseService");
+    event.setDateTime(now.toString());
     event.setTransactionId(UUID.randomUUID().toString());
     event.setType("CaseCreated");
     Address address = new Address();
-    address.setAddressLine1("1 Main Street");
-    address.setAddressLine2("Upper Upperingham");
-    address.setAddressLine2("Lower Lowerington");
-    address.setAddressType("CE");
-    address.setArid("XXXXX");
-    address.setCountry("E");
-    address.setEstabType("XXX");
-    address.setLatitude("50.863849");
-    address.setLongitude("-1.229710");
-    address.setPostcode("UP103UP");
-    address.setTownName("Royal Midtowncastlecesteringshirehampton-upon-Twiddlebottom");
+    address.setAddressLine1(caze.getAddressLine1());
+    address.setAddressLine2(caze.getAddressLine2());
+    address.setAddressLine3(caze.getAddressLine3());
+    address.setAddressType(caze.getAddressType());
+    address.setArid(caze.getArid());
+    address.setRegion(caze.getRgn().substring(0, 1));
+    address.setEstabType(caze.getEstabType());
+    address.setLatitude(caze.getLatitude());
+    address.setLongitude(caze.getLongitude());
+    address.setPostcode(caze.getPostcode());
+    address.setTownName(caze.getTownName());
     CollectionCase collectionCase = new CollectionCase();
-    collectionCase.setActionableFrom("2019-04-01T12:00Z");
+    collectionCase.setActionableFrom(now.toString());
```

Comment:
This is an assumption. We might need to set this to be the start of the collection exercise, which we don't have so we'll probably need to hard-code.